src/autocomplete/databases/yql/helpers.ts

```diff
@@ -20,6 +21,36 @@ export const tokenDictionary: TokenDictionary = {
 type AnyRuleInList = (rules: number | number[]) => boolean;
 type AllRulesInList = (rules: number[]) => boolean;
 
+function isFirstPreviousTokenOfType(
+    tokenStream: TokenStream,
+    dictionary: TokenDictionary,
+    tokenIndex: number,
+    tokenType: number,
+): boolean | undefined {
+    const firstIndex = tokenIndex - 1;
+
+    let token = tokenStream.get(firstIndex);
+
+    if (token.type === tokenType) {
+        return true;
+    }
+
+    let currentIndex = firstIndex - 1;
```

Comment:
I think we can make a do while loop here, so we don't have to check for first index manually